### PR TITLE
[agent-operator] Added podLabels to agent-operator deployment pods

### PR DESCRIPTION
**Pull request by praveenjindal62**
Signed-off-by: Praveen Jindal <praveenjindal62@gmail.com>
We are using agent-operator charts and wanted to manage pods using some custom labels on pods. agent-operator doesn't provide flexibility yet to use custom podLabels on deployment agent-operator. Added .Values.podLabels that can be applied to deployment agent-operator

### CODE REVIEW

**Comment by praveenjindal62** [comment]
HI @trevorwhitney , Do you think if you could look into this as well ? 

This is same as the pull request [1441](https://github.com/grafana/helm-charts/pull/1441) which I created for Loki-Canary

**Review by trevorwhitney — COMMENTED** [review]
LGTM, but since I don't work on the agent nor have I been involved in this chart much, I'll leave it to an agent-operator maintainer/team member to approve.

**Review by zanhsieh — APPROVED** [review]

**Pull request merged**
